File: app/src/main/java/com/example/adulting/jdata/dao/CardTypeDAO.java
package com.example.adulting.jdata.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;


import com.example.adulting.jdata.entity.CardType;

import java.util.List;

@Dao
public interface CardTypeDAO {
    @Insert
    void insert(CardType cardType);

    @Query("SELECT * FROM card_types")
    LiveData<List<CardType>> getAllCardTypes();

}

File: app/src/main/java/com/example/adulting/data/dao/CardTypeDAO.kt
package com.example.adulting.data.dao

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query
import com.example.adulting.data.entities.CardType

@Dao
interface CardTypeDAO {
    @Query("SELECT * FROM CardType")
    fun getAll(): List<CardType>

    @Insert
    fun insertAll(vararg  cardType: CardType)

    /*
    @Query("SELECT * FROM CardType WHERE cardTypeId IN (:)")
    fun loadAllByIds(userIds: IntArray): List<User>

    @Query("SELECT * FROM user WHERE first_name LIKE :first AND " +
            "last_name LIKE :last LIMIT 1")
    fun findByName(first: String, last: String): User

    @Insert
    fun insertAll(vararg users: User)

    @Delete
    fun delete(user: User)
    */

}
